internal/app/connectconformance/client_runner.go

```diff
@@ -36,29 +36,31 @@ type clientRunner interface {
 	sendRequest(req *conformancev1alpha1.ClientCompatRequest, whenDone func(string, *conformancev1alpha1.ClientCompatResponse, error)) error
 	closeSend()
 	waitForResponses() error
+
+	isRunning() bool
+	stop()
 }
 
 func runClient(ctx context.Context, start processStarter) (clientRunner, error) {
-	ctx, cancel := context.WithCancel(ctx)
```

Comment:
This extra context was unused, so no need dealing with the cancellation of it.

At one point, I plumbed this through to `consumeOutput`, to have it abort when the context was cancelled. But then that would cause a race between it consuming all of the program's output and the process being cancelled (the parent context gets cancelled asynchronously when we observe the program has terminated).

So we rely solely on seeing EOF for `consumeOutput` to return (or for the timeout to elapse), instead of cancelling this context.